Editor/MenuActions/Selection/SelectPathFaces.cs

```diff
@@ -0,0 +1,157 @@
+            }
+        }
+
+        private static int[] Dijkstra(int start, int end, ProBuilderMesh mesh)
+        {
+            HashSet<int> visited = new HashSet<int>();
+            HashSet<int> toVisit = new HashSet<int>();
+            lastWings = lastMesh == mesh ? lastWings : WingedEdge.GetWingedEdges(mesh, true);
+            int wingCount = lastWings.Count;
+
+            if (mesh != lastMesh)
+            {
+                lastFacesIndex.Clear();
+
+                for (int i = 0; i < mesh.facesInternal.Length; i++)
+                {
+                    lastFacesIndex.Add(mesh.facesInternal[i], i);
+                }
+            }
+
+            float[] weights = new float[wingCount];
+            int[] predecessors = new int[wingCount];
+
+            for (int i = 0; i < wingCount; i++)
+            {
+                weights[i] = float.MaxValue;
+            }
+
+            int current = start;
+            weights[current] = 0;
+            visited.Add(current);
+
+            // Construct the paths between the start face and every other faces
+            while (visited.Count < wingCount)
+            {
+                var currentWing = lastWings[current];
+                var otherWing = currentWing;
+                // Update the weight array for each face next to the current one
+                do
+                {
+                    var opposite = otherWing.opposite;
+                    if (opposite == null)
+                        continue;
+
+                    var idx = lastFacesIndex[opposite.face];
+                    var weight = GetWeight(current, idx, mesh);
+                    // Change the predecessor and weight if the new path found if shorter
+                    if (weights[current] + weight < weights[idx])
+                    {
+                        weights[idx] = weights[current] + weight;
+                        predecessors[idx] = current;
+                    }
+                    // Add the face to the ones we can visit next, if not yet visited
+                    if (!toVisit.Contains(idx) && !visited.Contains(idx))
+                    {
+                        toVisit.Add(idx);
+                    }
+
+                    otherWing = otherWing.next;
+
+                } while (otherWing != currentWing);
+
+                // Look for the next face to visit, choosing the one with less weight
+                double min = double.MaxValue;
+                foreach (var i in toVisit)
+                {
+                    if (weights[i] < min)
+                    {
+                        min = weights[i];
+                        current = i;
+                    }
+                }
+                visited.Add(current);
+                toVisit.Remove(current);
+            } 
+
+            return predecessors;
+        }
+
+        private static float GetWeight(int face1, int face2, ProBuilderMesh mesh)
+        {
+            float baseCost = 10f;
```

Comment:
The number in this part are arbitrary and will need to be changed after some real case testing